packages/language-support/src/autocompletion/completionCoreCompletions.ts

```diff
@@ -348,11 +349,16 @@ const parameterCompletions = (
       // If there is a preceding token and it's not empty, compute the suffix
       if (previousToken.type !== CypherLexer.SPACE) {
         const param = maybeInsertText.insertText ?? `$${paramName}`;
-        const suffix = isSuffix(previousToken.text, param);
-        // We need to have the insert text without the starting $ in VSCode,
-        // otherwise when we have 'RETURN $' and we get offered $param
-        // we would complete RETURN $$param, which is not what we want
-        maybeInsertText = suffix ? { insertText: paramName } : {};
+        const hasDollar =
+          previousToken.type === CypherLexer.DOLLAR ||
+          tokens.at(previousToken.tokenIndex - 1).type === CypherLexer.DOLLAR;
+        // If the $ symbol is already there, we need to have the insert
+        // text without the starting $ in VSCode, otherwise when we have
+        // 'RETURN $' and we get offered $param we would complete
+        // RETURN $$param, which is not what we want
+        maybeInsertText = hasDollar
+          ? { insertText: param.slice(1) }
+          : maybeInsertText;
```

Comment:
Readjusts this part to fix the completions on the VSCode side. If we have started the dollar part, we need the insert text to come without the `$`, otherwise we would end up with double dollars.